### PR TITLE
Refactor/cleanup legacy

### PR DESCRIPTION
**Pull request by rolfvdhulst**
Removing classes that were never used **and** are also likely not to be useful in the future, and fixing some minor warnings here and there.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -83,8 +76,9 @@ target_link_libraries(roboteam_utils
 target_compile_options(roboteam_utils PRIVATE
         --std=c++17
         -Wall
-#        -Werror
         -Wextra
+        -pedantic-errors
+        -Werror=pedantic
```
Comment (Yuhanun) [inline CMakeLists.txt]:
```suggestion
        -pedantic-errors
        -Werror=pedantic
```

Don't you mean
-Werror 
-pedantic
?

**Review by Yuhanun — APPROVED** [review]
lgtm

File: CMakeLists.txt (hunk 1 above)
Comment (rolfvdhulst) [reply CMakeLists.txt]: Ah yeah, I was testing some gcc flags. I looked at the doc and -Werror -pedantic makes more sense.
These things are NOT the same though, see https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#Warning-Options

Comment (rolfvdhulst) [reply CMakeLists.txt]: As in, these things are legitimate flags, but I meant to change them back.

**Comment by rolfvdhulst** [comment]
I double-checked all other repositories for compliation. This should be fine.

**Comment by rolfvdhulst** [comment]
DONT MERGE FOR NOW.

there is some issues.

**Pull request merged**